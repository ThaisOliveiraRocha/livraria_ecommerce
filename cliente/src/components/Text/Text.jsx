import React from "react";
import PropTypes from "prop-types";
import { TextComponent } from "./Text.styles";

const Text = ({ font, children, color, bold }) => {
  return (
    <TextComponent font={font} color={color} bold={bold}>
      {children}
    </TextComponent>
  );
};

Text.propTypes = {
  // bla: PropTypes.string,
};

Text.defaultProps = {
  // bla: 'test',
};

export default Text;
