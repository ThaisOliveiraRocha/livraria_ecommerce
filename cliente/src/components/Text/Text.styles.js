import styled from 'styled-components';

export const TextComponent = styled.span`
    font-size: 14px;
    font-family: Impact, Haettenschweiler, "Arial Narrow Bold", sans-serif;
    color: ${({ color }) => (color ? color : "black")};
`;
