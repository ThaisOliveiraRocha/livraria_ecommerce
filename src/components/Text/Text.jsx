import React from 'react';
import PropTypes from 'prop-types';
import { TextComponent } from './Text.styles';

const Text = ({children, color}) => (
  <TextComponent color={color}>{children}</TextComponent>
);

Text.propTypes = {
  // bla: PropTypes.string,
};

Text.defaultProps = {
  // bla: 'test',
};

export default Text;
