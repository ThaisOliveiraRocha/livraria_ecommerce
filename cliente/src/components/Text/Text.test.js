import React from "react";
import { shallow } from "enzyme";
import Text from "./Text";

describe("<Text />", () => {
  it("should render without crashing", () => {
    shallow(<Text />);
  });
  it("should render basic infos", () => {
    const wrapper = shallow(<Text />);
    expect(wrapper).toMatchSnapshot();
  });
  
});
