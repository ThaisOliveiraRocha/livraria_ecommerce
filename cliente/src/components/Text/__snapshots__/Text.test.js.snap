// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Text /> should render basic infos 1`] = `<styled.span />`;
